lib/cartoRenderer.js

```diff
@@ -1073,10 +1073,32 @@ var cartoSelectorIsMatch = function(definition, feature, source, zoom) {
     for (var filterKey in definition.filters) {
       var filter = definition.filters[filterKey];
       
-      if (filter.op === "=") {
+      switch (filter.op) {
+      case "=":
         if (feature.properties[filter.key] !== filter.val) {
           return false;
         }
+        break;
+      case "<":
+        if (feature.properties[filter.key] >= filter.val) {
```

Comment:
The filters work by returning early from the property loop based on a single `false` match, rather than iterating over the entire set of properties to ensure they all are `true`. 

I don't know if it would be more performant to use a true functional method, but it's following the pattern of the previous implementation of the `=` matcher.